subprojects/wrapper/src/main/java/org/gradle/wrapper/Install.java

```diff
@@ -259,6 +271,30 @@ private void unzip(File zip, File dest) throws IOException {
         }
     }
 
+    /**
+     * Checks the entry name for zip-slip vulnerable sequences.
+     *
+     * <b>IMPLEMENTATION NOTE</b>
+     * We do it this way instead of the way recommended in https://snyk.io/research/zip-slip-vulnerability
+     * for performance reasons, calling {@link File#getCanonicalPath()} is too expensive.
+     *
+     * @throws IllegalArgumentException if the entry contains vulnerable sequences
+     */
+    private static String safeZipEntryName(String name) {
+        if (isUnsafeZipEntryName(name)) {
+            throw new IllegalArgumentException(format("'%s' is not a safe zip entry name.", name));
+        }
+        return name;
+    }
+
+    private static boolean isUnsafeZipEntryName(String name) {
+        return name.isEmpty()
+            || name.startsWith("/")
+            || name.startsWith("\\")
+            || name.contains("..")
+            || name.contains(":");
```

Comment:
This check is now different than the one in `GUtil`, rejecting names with `:` on unices. It should be mostly fine when it comes to Gradle distributions though. Custom distros with such files used only on unices would fail to unpack. It's a corner case but symmetry would have been nice for both users and maintainers of this code, for the least surprise.

Reply:
Right, symmetry wouldn't cost much here.

Do you think it would be worth introducing a separate project so the implementation could be shared between the wrapper and the distribution? Currently they only share the `:cli` implementation.

Reply:
At the very least we could make the implementation symmetric and cross reference the duplicated code for future maintainers. Adding a separate project/jar for that function feels overkill but also the right thing to do. It'd be a future proof way to maintain code shared with the wrapper that is not about `:cli`.